mkosi/config.py

```diff
@@ -1160,6 +1161,27 @@ def config_parse_minimum_version(value: Optional[str], old: Optional[GenericVers
     return max(old, new)
 
 
+def config_path_run(script: Path) -> str:
+    script = script.absolute()
+
+    if not os.access(script, os.X_OK):
+        die(f"{script} is not executable")
+
+    return run([script], stdout=subprocess.PIPE, env=os.environ).stdout.rstrip("\n")
+
+
+def config_path_read(file: Path) -> str:
+    return file.read_text().rstrip("\n")
+
+
+def config_path_run_or_read(file: Path) -> str:
+    "Run the specified file and capture its output if it's executable, else read file contents"
+    if os.access(file, os.X_OK):
+        return config_path_run(file)
+    else:
+        return config_path_read(file)
```

Comment:
Nit:
```suggestion
    
    return config_path_read(file)
```

Also, this is in theory a backwards incompatible change, but I'm not sure we should read the whole file here. Having more than a single line in a version doesn't make sense conceptually. My point is, that it's easy to forget to make a file executable and ending up with the text of a script wouldn't be helpful, instead we should die if the text in this case starts with `#!/`, since neither character is allowed in the version format spec and I'm not aware it is used in any version string. If people really want that a multiline script as a version, they can write the script and just add a `head -n -1 $0` at the end.

At the very least we should log a warning.